### ISSUE TITLE
Filters option of Relationship field does not work as expected

### ISSUE THREAD

**Issue by hmlamad** [keystonejs/keystone#4206]
### Expected behavior

The filters option of relationship field doesn't work as expected.
In the keystone CMS, I try to create a document of MessageModel, and in the create form, if I choose push type to be "Topic" and application selected as let's say "Application-A". Then all the topics in the create form are supposed to be filtered.
```javascript
ApplicationModel.add({
    name: {type: Types.Text, index: true, required: true, initial: true},
});

MessageModel.add({
    pushType: {type: Types.Select, options: ['application', 'topic', 'device']},
    application: {type: Types.Relationship, ref: 'Application'},
    topic: {type: Types.Relationship, ref: 'Topic', dependsOn: {pushType: 'topic'}, filters: { application: ':application' }},
    device: {type: Types.Relationship, ref: 'Device',  dependsOn: {pushType: 'device',}, filters: { application: ':application' }},
});

DeviceModel.add({
    application: {type: Types.Relationship, ref: 'Application'},
});

TopicModel.add({
    application: {type: Types.Relationship, ref: 'Application'},
});
```

### Actual/Current behavior

However, the device or topic field did not filter as my expectation.

### Steps to reproduce the actual/current behavior

if I select push type then application, topic will not filter
![screen shot 2017-04-10 at 8 36 34 pm](https://cloud.githubusercontent.com/assets/10290082/24862099/d497f79e-1e2d-11e7-8cf7-3d5483069333.png)

if i select application first then push type
![screen shot 2017-04-10 at 8 36 47 pm](https://cloud.githubusercontent.com/assets/10290082/24862145/07fbebf4-1e2e-11e7-9361-626803416181.png)
then i change application again
![screen shot 2017-04-10 at 8 42 08 pm](https://cloud.githubusercontent.com/assets/10290082/24862207/3ed04314-1e2e-11e7-9992-fc78c04dbcfb.png)

I have checked that the loadOptions function in relationship field does not execute upon the change of application field

P.S. I have not checked the behaviour in EditForm but I guess the performance is the same
### Environment

| Software         | Version
| ---------------- | -------
| Keystone         | v4.0.0-beta.5
| Node             | v6.7.0

**Comment by metinmobven** [comment]
+1

**Comment by export-mike** [comment]
use at your own risk until its merged. ```yarn add @thinkmill/keystone@4.0.0-beta.5-jm-fork```

**Comment by Noviny** [comment]
It has been merged!

**Comment by kevrat** [comment]
Looks like it has been revert? I'm on beta7 branch and this does not work

**Comment by sheldon-sminq** [comment]
Yes does not work for beta7 branch

**Comment by dufemeng** [comment]
Did this solved? I have the same question.

**Comment by dufemeng** [comment]
> Looks like it has been revert? I'm on beta7 branch and this does not work

+1 . How did you solved ?

**Comment by dragosstancu** [comment]
Filters only if the filtering field is set to initial. Then it doesn't work anymore.

**Comment by VinayaSathyanarayana** [comment]
@dragosstancu - Please provide additional details (preferably with an example). 
It is not working in Beta8 and also in released 4.0.0
